File: Java/BasicSample/app/src/main/java/com/neyogiry/android/sample/DateModel.java
package com.neyogiry.android.sample;

import java.util.ArrayList;

public class DateModel {

    ArrayList<String> mDates = new ArrayList<>();

    public DateModel() {
    }

    public void saveDate(String date, DateInterfaces.Model callback) {
        try{
            mDates.add(0, date);
            onSuccess(callback);
        }catch (Exception e){
            onError(callback, e.toString());
        }
    }

    public void onSuccess(DateInterfaces.Model callback) {
        callback.onResponse(mDates);
    }

    public void onError(DateInterfaces.Model callback, String error) {
        callback.onError(error);
    }

}

File: Kotlin/BasicSample/app/src/main/java/com/neyogiry/android/sample/DateModel.kt
package com.neyogiry.android.sample

class DateModel{

    var mDates: ArrayList<String> = ArrayList()

    fun saveDate(date: String, callback: DateInterfaces.Model) {
        try {
            mDates.add(0, date)
            onSuccess(callback)
        } catch (e: Exception) {
            onError(callback, e.toString())
        }

    }

    private fun onSuccess(callback: DateInterfaces.Model) {
        callback.onResponse(mDates)
    }

    private fun onError(callback: DateInterfaces.Model, error: String) {
        callback.onError(error)
    }

}